### PR TITLE
SPLICE-2056 Exception handling for connection exceptions added

### PR DESCRIPTION
**Pull request by ipraznik-splice**

### CODE REVIEW

File: hbase_sql/src/main/java/com/splicemachine/stream/RemoteQueryClientImpl.java [hunk 1]
```diff
@@ -113,7 +114,12 @@ public void run() {
                         streamListener.completed(olapResult);
                     } catch (ExecutionException e) {
                         LOG.warn("Execution failed", e);
-                        streamListener.failed(e.getCause());
+                        Throwable cause = e.getCause();
+                        if (cause instanceof IOException || cause instanceof ConnectException) {
+                            streamListener.failed(StandardException.newException(SQLState.OLAP_SERVER_CONNECTION, cause));
```
Comment (martinrupp) [inline hbase_sql/src/main/java/com/splicemachine/stream/RemoteQueryClientImpl.java:119]: not sure if the exception is clear in the log, we could differentiate the log we print for these cases, e.g.

```
Throwable cause = e.getCause();
if (cause instanceof IOException || cause instanceof ConnectException) {
  LOG.warn("Execution failed: Lost OLAP Connection", e);
  streamListener.failed(StandardException.newException(SQLState.OLAP_SERVER_CONNECTION, cause));
} else {
  LOG.warn("Execution failed: Lost OLAP Connection", e);
  streamListener.failed(cause);
}
```

Comment (ipraznik-splice) [reply hbase_sql/src/main/java/com/splicemachine/stream/RemoteQueryClientImpl.java:119]: In this case in the log you will see the wrapper exception with the code SQLState.OLAP_SERVER_CONNECTION, that is used fo such cases, and of course stack trace to navigate to the place

**Comment by ipraznik-splice** [comment]
jenkins please test branch @cdh6.3.0,skipTestsLongerThan2Minutes

**Comment by spliceci** [comment]
TEST FAILED

**Comment by ipraznik-splice** [comment]
jenkins please test branch @cdh6.3.0,skipTestsLongerThan2Minutes

**Comment by spliceci** [comment]
TEST FAILED

**Comment by ipraznik-splice** [comment]
jenkins please test branch @cdh6.3.0,skipTestsLongerThan2Minutes

**Comment by spliceci** [comment]
TEST SUCCEEDED +1

**Pull request merged**